### PR TITLE
cqrlog: init at 2.5.2

### LINKED ISSUE #111164: cqrlog
**Project description**
CQRLOG is an advanced ham radio logger based on MySQL database. Provides radio control based on hamlib libraries (currently support of 140+ radio types and models), DX cluster connection, online callbook, a grayliner, internal QSL manager database support and a most accurate country resolution algorithm based on country tables created by OK1RR. CQRLOG is intended for daily general logging of HF, CW & SSB contacts and strongly focused to easy operation and maintenance. 

**Metadata**

* homepage URL: https://www.cqrlog.com/
* source URL: https://github.com/ok2cqr/cqrlog
* license: GPL-2.0 License 
* platforms: unix, linux

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
Closes https://github.com/NixOS/nixpkgs/issues/111164

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Need to configure the mysql db for further testing.

### CODE REVIEW

File: pkgs/applications/radio/cqrlog/default.nix [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+, gdk-pixbuf
+, glib
+, gtk2-x11
+, libX11
+, pango
+
+, hamlib
+, mysql57
+, tqsl
+, xdg_utils
```
Comment (cole-h) [inline pkgs/applications/radio/cqrlog/default.nix:17]: `xdg_utils` is an alias to `xdg-utils`.

```suggestion
, xdg-utils
```

File: pkgs/applications/radio/cqrlog/default.nix [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+, lazarus
+, atk
+, cairo
+, gdk-pixbuf
+, glib
+, gtk2-x11
+, libX11
+, pango
+
+, hamlib
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/cqrlog/default.nix:14]:
```suggestion
, pango
, hamlib
```

File: pkgs/applications/radio/cqrlog/default.nix [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+, mysql57
+, tqsl
+, xdg_utils
+, xplanet
+
+, autoPatchelfHook
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/cqrlog/default.nix:20]:
```suggestion
, xplanet
, autoPatchelfHook
```

File: pkgs/applications/radio/cqrlog/default.nix [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    glib
+    gtk2-x11
+    libX11
+    pango
+  ];
+  propagatedBuildInputs = [
+    hamlib
+    mysql57
+    tqsl
+    xdg_utils
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/cqrlog/default.nix:78]:
```suggestion
    xdg-utils
```

File: pkgs/applications/radio/cqrlog/default.nix [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+  makeFlags = [
+    "FPC=fpc"
+    "PP=fpc"
+    "DESTDIR=$(out)"
+  ];
+
+  postFixup = ''
+    # Needed for sqldb to find and load libmysqlclient
+    patchelf --add-needed "${mysql57}/lib/libmysqlclient.so.20" "$out/bin/.cqrlog-wrapped"
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/cqrlog/default.nix:90]: Can we get the 20 dynamically?

File: pkgs/applications/radio/cqrlog/default.nix [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Linux logging program for amateur radio operators";
+    homepage = "https://www.cqrlog.com/";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ shamilton ];
+    platforms = lib.platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/cqrlog/default.nix:98]:
```suggestion
    platforms = platforms.linux;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/radio/cqrlog/default.nix (hunk 2 above)
Comment (cole-h) [reply pkgs/applications/radio/cqrlog/default.nix:14]: Does the presence (or absence) of newlines really matter?

Comment (SCOTT-HAMILTON) [reply pkgs/applications/radio/cqrlog/default.nix:14]: This is done to separate buildInputs from propagatedBuildInputs.

File: pkgs/applications/radio/cqrlog/default.nix (hunk 3 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/radio/cqrlog/default.nix:20]: autoPatchelfHook isn't a propagatedBuildInputs, thus the space.

File: pkgs/applications/radio/cqrlog/default.nix (hunk 1 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/radio/cqrlog/default.nix]: Fixed

File: pkgs/applications/radio/cqrlog/default.nix (hunk 4 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/radio/cqrlog/default.nix]: Fixed.

File: pkgs/applications/radio/cqrlog/default.nix (hunk 6 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/radio/cqrlog/default.nix]: Fixed.

File: pkgs/applications/radio/cqrlog/default.nix (hunk 5 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/radio/cqrlog/default.nix]: I don't know if this was what you were looking for https://github.com/NixOS/nixpkgs/blob/64890ee669b7e41d0761348e3961dd0186b1d2ca/pkgs/applications/radio/cqrlog/default.nix#L90-L92

File: pkgs/applications/radio/cqrlog/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/radio/cqrlog/default.nix:14]: Over time this will bitrot anyway and be confined.

Comment (SCOTT-HAMILTON) [reply pkgs/applications/radio/cqrlog/default.nix]: Fixed.

File: pkgs/applications/radio/cqrlog/default.nix (hunk 3 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/radio/cqrlog/default.nix]: Fixed.

File: pkgs/applications/radio/cqrlog/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/applications/radio/cqrlog/default.nix]: Yeah, we probably want to use it or this package breaks on the next abi bump of libmysqlclient

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113228` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cqrlog</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
cqrlog:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern '/usr/bin/mysqld_safe' doesn't match anything in file 'src/dData.pas'
```

Please check the offending substituteInPlace for typos or changes in source.

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/radio/cqrlog/default.nix:33:15:

```
   |
33 |   patches = [ ./fix-makefile-lazarusdir.patch ];
   |               ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
</details>

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -21516,6 +21516,8 @@ in
     python3Packages = python37Packages;
   };
 
+  cqrlog = callPackage ../applications/radio/cqrlog {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:21519]:
```suggestion
  cqrlog = callPackage ../applications/radio/cqrlog { };
```

File: pkgs/applications/radio/cqrlog/default.nix [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  makeFlags = [
+    "FPC=fpc"
+    "PP=fpc"
+    "DESTDIR=$(out)"
+  ];
+
+  postFixup = ''
+    # Needed for sqldb to find and load libmysqlclient
+    libmysqlclient=$(find "${mysql57}/lib" -name "libmysqlclient.so.*" | tail -n1)
+    echo "Adding \`$libmysqlclient\` to binary..."
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/cqrlog/default.nix:92]:
```suggestion
```

File: pkgs/applications/radio/cqrlog/default.nix [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+  ];
+
+  makeFlags = [
+    "FPC=fpc"
+    "PP=fpc"
+    "DESTDIR=$(out)"
+  ];
+
+  postFixup = ''
+    # Needed for sqldb to find and load libmysqlclient
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/cqrlog/default.nix:90]:
```suggestion
```

**Pull request merged**